internal/data/metric.go

```diff
@@ -122,66 +237,171 @@ type internalMetric struct {
 	doubleDataPoints    []DoubleDataPoint
 	histogramDataPoints []HistogramDataPoint
 	summaryDataPoints   []SummaryDataPoint
+	// True when the slice was replace.
```

Comment:
Which slice change does this track? I assume this is set to try when any of the slices is changed?

Reply:
any slice, optimization can be done to track every slice independently, I didn't try to optimize too much now, in order to get something out soon.

Reply:
Sounds good. I agree, let's have the first version out. We can improve later.